sui_core/src/authority_server.rs

```diff
@@ -138,7 +144,59 @@ impl AuthorityServer {
                         continue;
                     }
 
-                    let response = BatchInfoResponseItem(item);
+                    // add tx_info if requested
+                    let mut response = BatchInfoResponseItem(item.clone());
+                    if include_tx_info {
```

Comment:
This does reads to the DB for each item, and for each stream -- this is going to kill performance both for the streaming and possibly everything else. Right now our DB is the resource we are choked on. Now that I see this, I am even more inclined to remove the `include_tx_info` and have the clients pull the transactions info when needed.